spec/support/utils.rb

```diff
@@ -1,4 +1,49 @@
+require 'base64'
+
 module Utils
+  # A limited extended JSON parser to use in FLE specs.
+  # DO NOT consider this a complete extended JSON parser.
+  # This method will be removed once a complete extended
+  # JSON parser has been implemented.
+  #
+  # This method may modify the provided argument
+  def parse_extended_json(doc)
+    requires_parsing = doc.is_a?(Hash) && (doc.key?('$binary') || doc.key?('$date') || doc.key?('$numberInt'))
+    return parse_json_object(doc) if requires_parsing
+
+    if doc.is_a?(Hash)
+      return doc.each do |key, value|
+        doc[key] = parse_extended_json(value)
+      end
+    end
+
+    if doc.is_a?(Array)
+      return doc.each.with_index do |el, idx|
+        doc[idx] = parse_extended_json(el)
+      end
+    end
+
+    return doc
+  end
+  module_function :parse_extended_json
```

Comment:
It would be really helpful for me to have a limited subset of extended JSON parsing to load data keys/schema maps for the FLE spec. Let me know if you think I should approach this some other way.